src/Oriflame.RedisMessaging.ReliableDelivery/Subscribe/Validation/ValidationResultForMissingMessages.cs

```diff
@@ -16,5 +18,31 @@ public override string ToString()
         {
             return $"MissingMessages:LastProcessedMessageId={LastProcessedMessageId}";
         }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            return obj is ValidationResultForMissingMessages other && Equals(other);
+        }
+
+        public bool Equals(ValidationResultForMissingMessages other)
+        {
+            return LastProcessedMessageId == other.LastProcessedMessageId;
+        }
+
+        public override int GetHashCode()
+        {
+            return LastProcessedMessageId.GetHashCode();
+        }
+
+        public static bool operator ==(ValidationResultForMissingMessages left, ValidationResultForMissingMessages right)
+        {
+            return left.Equals(right);

```

Comment:
"left" could not be null? (the same below)

Reply:
Since type `ValidationResultForMissingMessages` is struct it cannot be null.